File: Android/dokit/src/main/java/com/didichuxing/doraemonkit/widget/bottomview/AssociationView.java
package com.didichuxing.doraemonkit.widget.bottomview;

import android.view.View;

/**
 * 可提交的view
 *
 * @author vinda
 * @since 15/5/21
 */
public abstract class AssociationView {

    private OnStateChangeListener onStateChangeListener;

    /**
     * 提交
     */
    public abstract Object submit();

    /**
     * 取消
     */
    public abstract void cancel();

    /**
     * 获取视图
     *
     * @return
     */
    public abstract View getView();

    /**
     * 能否提交
     *
     * @return
     */
    public abstract boolean isCanSubmit();

    public abstract void onShow();

    public abstract void onHide();


    final void setOnStateChangeListener(OnStateChangeListener listener) {
        onStateChangeListener = listener;
    }

    final OnStateChangeListener getOnStateChangeListener() {
        return onStateChangeListener;
    }


    interface OnStateChangeListener {
        void onStateChanged();
    }
}

File: Android/kotlin/doraemonkit/src/main/java/com/didichuxing/doraemonkit/widget/bottomview/AssociationView.kt
package com.didichuxing.doraemonkit.widget.bottomview

import android.view.View

/**
 * 可提交的view
 *
 * @author vinda
 * @since 15/5/21
 */
abstract class AssociationView {
    var onStateChangeListener: OnStateChangeListener? = null

    /**
     * 提交
     */
    abstract fun submit(): Any

    /**
     * 取消
     */
    abstract fun cancel()

    /**
     * 获取视图
     *
     * @return
     */
    abstract val view: View?

    /**
     * 能否提交
     *
     * @return
     */
    abstract val isCanSubmit: Boolean
    abstract fun onShow()
    abstract fun onHide()

    interface OnStateChangeListener {
        fun onStateChanged()
    }
}